File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/OneclickEchoRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

@JsonInclude(value = JsonInclude.Include.NON_NULL)
@RequiredArgsConstructor
public class OneclickEchoRequest extends SignBase {
    @NonNull
    @Getter
    private String merchantId;
    @NonNull
    @Getter
    private String origPayId;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, origPayId);
        add(sb, dttm);
        deleteLast(sb);
        return sb.toString();
    }

}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/OneclickEchoRequest.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class OneclickEchoRequest : SignBaseRequest //OK
{
    public OneclickEchoRequest()
    {
    }

    public OneclickEchoRequest(string? merchantId, string? OrigPayId)
    {
        MerchantId = merchantId;
        this.OrigPayId = OrigPayId;
    }

    [JsonProperty("origPayId", Required = Required.Always)]
    public string? OrigPayId { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, OrigPayId);
        Add(sb, Dttm);
        DeleteLast(sb);
        return sb.ToString();
    }
}